ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -206,6 +208,14 @@ public CreateSourceAsProperties getProperties() {
     return withProperties;
   }
 
+  void addTableFunction(final FunctionCall functionCall) {
+    this.tableFunctions.add(functionCall);
```

Comment:
add a sanity null check to functionCall given that we're storing it.

(From https://github.com/confluentinc/ksql/pull/3589#discussion_r336428831)

Reply:
It makes me cringe to add this, but I will for now, at least until the discussion on project style is complete

Reply:
done